### PR TITLE
CSS/JS naming update

### PR DESCRIPTION
**Pull request by mrmohtas**
## What? 

This is an update to the guidelines to incorporate the latest team agreement on how we should handle CSS class naming and how we are going to add JS behaviour hooks to HTML elements.

### CODE REVIEW

File: sass-css.md [hunk 1]
```diff
@@ -1,33 +1,57 @@
...
+
+## Preamble: the TL,DR
 
 On a large site the most important thing to understand is to strive for re-use as much as possible.
 
 With that end the overall strategy of our SASS/CSS is to take some well understood (and battle-tested) functionality from bootstrap and add our components on top.
 
-> With great power, comes great responsibility.
-> --Uncle Ben (the Spiderman one, not the rice one)
+- We prefer the indented SASS syntax
```
Comment (andycroll) [inline sass-css.md]: We do? Snuck that one through @mrmohtas... I don't really care, but ideally we shouldn't be indenting at all if we can avoid it.

File: sass-css.md [hunk 2]
```diff
@@ -238,52 +231,189 @@ So remember, do not fear multiple classes on elements.
...
+
+**[[⬆]](#TOC)**
+
+## <a name="housetrip-style">HouseTrip Style</a>
+
+We use a tweaked bootstrap (v3) grid to layout our pages.
+
+The [bootstrap documentation](http://getbootstrap.com/css/#grid) is the best place to read up on how that works. We can use all of the existing types of bootstrap grid classes: e.g. `col-sm-offset-X`, `.col-sm-push-3`.
+
+It uses (currently) only the `.col-sm-X` classes for the desktop version, we are not ready for full responsivity in our designs, but by following these guidelines _we can get there_.
```
Comment (andycroll) [inline sass-css.md]: Think we're pretty much there...

File: sass-css.md [hunk 3]
```diff
@@ -238,52 +231,189 @@ So remember, do not fear multiple classes on elements.
...
+
+We can also inline SVG to reduce HTTP requests as we move away from any support other than IE8.
+
+**A note on icon-fonts**
+
+Icon fonts are great. Including *all the icons* is not great. Generating classes for all the icons you need is a recipie for exventual CSS bloat.
+
+We currently have the glyphicons fonts included in v4, but we use only a handful for their stated purpose (tiny icons next to text) and spend a lot of time in CSS tweaking their position.
+
+Whilst icon fonts are not _off the table_ for simplicity let's stick with SVG for now.
```
Comment (andycroll) [inline sass-css.md]: I think industry experience is beginning to show it's _definitely_ a bad idea.

File: sass-css.md [hunk 4]
```diff
@@ -238,52 +231,189 @@ So remember, do not fear multiple classes on elements.
...
+
+For all non-photographic images we should try and use SVG. It is XML-based and therefore textand this compresses extremely well.
+
+We can also inline SVG to reduce HTTP requests as we move away from any support other than IE8.
+
+**A note on icon-fonts**
+
+Icon fonts are great. Including *all the icons* is not great. Generating classes for all the icons you need is a recipie for exventual CSS bloat.
+
+We currently have the glyphicons fonts included in v4, but we use only a handful for their stated purpose (tiny icons next to text) and spend a lot of time in CSS tweaking their position.
```
Comment (andycroll) [inline sass-css.md]: And I think in the responsive... it'd be great to get them out.

File: sass-css.md (hunk 1 above)
Comment (mrmohtas) [inline sass-css.md]: Most of this is reshuffling of the old guidelines to give it more order. I didn't add this

File: sass-css.md [hunk 5]
```diff
@@ -238,52 +231,189 @@ So remember, do not fear multiple classes on elements.
...
+
+CSS `#something` values have very high [specificity](http://www.w3.org/TR/selectors/#specificity). Style with classes. In fact, use `#id`s as sparingly as possible.
+
+**[[⬆]](#TOC)**
+
+### Do not style JavaScript classes
+
+**This is the old way of adding JS (see [JS naming](#js-naming))**
+
+We have a convention of using `.js-hook-for-behaviour` classes to target JS. Don't style these, even if if means creating 'parallel' classes for styling.
```
Comment (andycroll) [inline sass-css.md]: Past tense required.

File: sass-css.md (hunk 2 above)
Comment (mrmohtas) [inline sass-css.md]: I missed this, I'll update

File: sass-css.md [hunk 6]
```diff
@@ -526,41 +567,18 @@ The notable exception being, of course, that most of your layout will be done wi
 
 Avoid using floats, padding, and margins in emails.
 
-## Javascript
+**[[⬆]](#TOC)**
+
+### Javascript
```
Comment (andycroll) [inline sass-css.md]: This section needs updating to be in line with the discussed approach?

File: sass-css.md (hunk 3 above)
Comment (mrmohtas) [inline sass-css.md]: Agreed

File: sass-css.md (hunk 1 above)
Comment (andycroll) [inline sass-css.md]: But while you're here...

Comment (dncrht) [inline sass-css.md]: I'd like to know what's "Indented SASS syntax"

**Comment by mrmohtas** [comment]
All points should be addressed now

File: sass-css.md (hunk 4 above)
Comment (mrmohtas) [inline sass-css.md]: Without v4 they have no place. I think only PPP content navigation is still using them. 

I'll do some digging to detail where we are still using them

Comment (dncrht) [inline sass-css.md]: AFAIK, glyphicons are used in PPP amenities.

**Comment by andycroll** [comment]
:+1:

File: sass-css.md (hunk 4 above)
Comment (mrmohtas) [inline sass-css.md]: Anders wants to redo those, will push that case when I redo the host dashboard and get rid of the last of the v4 layouts

**Pull request merged**